packages/react-native-sdk/src/components/PermissionsNotification.tsx

```diff
@@ -0,0 +1,109 @@
+import { AxiosError, OwnCapability } from '@stream-io/video-client';
+import { useCall, useHasPermissions } from '@stream-io/video-react-bindings';
+import {
+  PropsWithChildren,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { Alert, Pressable } from 'react-native';
+
+export type PermissionNotificationProps = PropsWithChildren<{
+  /**
+   * The permission to check for.
+   */
+  permission: OwnCapability;
+
+  /**
+   * The message to display in the notification once
+   * the requested permission is granted.
+   */
+  messageApproved: string;
+
+  /**
+   * The message to display in the notification once a permission
+   * is revoked.
+   */
+  messageRevoked: string;
+
+  /**
+   * The message to display in the notification while
+   * the requested permission is awaiting approval.
+   */
+  messageAwaitingApproval: string;
+}>;
+
+export const PermissionNotification = (props: PermissionNotificationProps) => {
+  const {
+    permission,
+    messageApproved,
+    messageRevoked,
+    messageAwaitingApproval,
+    children,
+  } = props;
+  const hasPermission = useHasPermissions(permission);
+  const previousHasPermission = useRef(hasPermission);
+
+  const [isAwaitingApproval, setIsAwaitingApproval] = useState(false);
+  const call = useCall();
+
+  const showGrantedNotification = useCallback(() => {
+    Alert.alert(messageApproved);
+  }, [messageApproved]);
+
+  const showRevokedNotification = useCallback(() => {
+    Alert.alert(messageRevoked);
+  }, [messageRevoked]);
+
+  const showApprovalRequestNotification = useCallback(() => {
+    Alert.alert(messageAwaitingApproval);
+  }, [messageAwaitingApproval]);
```

Comment:
callbacks unneeded here IMHO

Reply:
Umm, these are a dependency to the useEffect below, so not having a callback actually throws an eslint warning of:
```
The 'showGrantedNotification' function makes the dependencies of useEffect Hook (at line 78) change on every render. Move it inside the useEffect callback. Alternatively, wrap the definition of 'showGrantedNotification' in its own useCallback() Hook.
```

Should I remove the function from the useEffect dependency and add an eslint flag that would stop the warning? @vanGalilea

Reply:
Also, I see no harm in having the callback, since its dependent on prop only.